### ISSUE TITLE
MySQL 5.7 default mode causes 500 errors

### ISSUE THREAD

**Issue by FloatingGhost** [MISP/MISP#1732]
Pretty simple one, the default sql_mode on mysql5.7 includes "ONLY_FULL_GROUP_BY" -- this causes an incompatibility with something. I'll give the error below, but it's pretty easily fixed by 
```
 SET GLOBAL sql_mode=(SELECT REPLACE(@@sql_mode,'ONLY_FULL_GROUP_BY',''));
```

I'd fix it and PR, but I've no idea where would be best to put it :confused: 

### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Bug
| OS version (server)       | Ubuntu 16.04
| OS version (client)       | Ubuntu 16.04
| PHP version               | 7.0
| MISP version / git hash   | 2.4.56, b653a2aab
| Browser                   | Firefox

### Expected behavior

Click on event ID -> Show Event

### Actual behavior

500 Internal server

### Steps to reproduce the behavior

Add ONLY_FULL_GROUP_BY to sql_mode, then  try to view an event.

### Logs, screenshots, configuration dump, ...

```
2016-12-08 12:37:53 Error: [PDOException] SQLSTATE[42000]: Syntax error or access violation: 1055 Expression #1 of SELECT list is not in GROUP BY clause and contains nonaggregated column 'misp.GalaxyCluster.id' which is not functionally dependent on columns in GROUP BY clause; this is incompatible with sql_mode=only_full_group_by
Request URL: /events/view/126
Stack Trace:
#0 /var/www/MISP/app/Lib/cakephp/lib/Cake/Model/Datasource/DboSource.php(468): PDOStatement->execute(Array)
#1 /var/www/MISP/app/Lib/cakephp/lib/Cake/Model/Datasource/DboSource.php(434): DboSource->_execute('SELECT `GalaxyC...', Array)
#2 /var/www/MISP/app/Lib/cakephp/lib/Cake/Model/Datasource/DboSource.php(671): DboSource->execute('SELECT `GalaxyC...', Array, Array)
#3 /var/www/MISP/app/Lib/cakephp/lib/Cake/Model/Datasource/DboSource.php(1120): DboSource->fetchAll('SELECT `GalaxyC...', Array)
#4 /var/www/MISP/app/Lib/cakephp/lib/Cake/Model/Model.php(3038): DboSource->read(Object(GalaxyCluster), Array)
#5 /var/www/MISP/app/Lib/cakephp/lib/Cake/Model/Model.php(3010): Model->_readDataSource('list', Array)
#6 /var/www/MISP/app/Controller/EventsController.php(819): Model->find('list', Array)
#7 /var/www/MISP/app/Controller/EventsController.php(901): EventsController->__viewUI(Array, false, NULL)
#8 [internal function]: EventsController->view('126')
#9 /var/www/MISP/app/Lib/cakephp/lib/Cake/Controller/Controller.php(491): ReflectionMethod->invokeArgs(Object(EventsController), Array)
#10 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(193): Controller->invokeAction(Object(CakeRequest))
#11 /var/www/MISP/app/Lib/cakephp/lib/Cake/Routing/Dispatcher.php(167): Dispatcher->_invoke(Object(EventsController), Object(CakeRequest))
#12 /var/www/MISP/app/webroot/index.php(92): Dispatcher->dispatch(Object(CakeRequest), Object(CakeResponse))
#13 {main}
```

**Comment by FloatingGhost** [comment]
hm, another potentially 5.7 issue
```
2016-12-08 12:54:50 Error: [PDOException] SQLSTATE[HY000]: General error: 1364 Field 'description' doesn't have a default value
Request URL: /galaxies/update
Stack Trace:
#0 /var/www/MISP/app/Lib/cakephp/lib/Cake/Model/Datasource/DboSource.php(468): PDOStatement->execute(Array)
#1 /var/www/MISP/app/Lib/cakephp/lib/Cake/Model/Datasource/DboSource.php(434): DboSource->_execute('INSERT INTO `mi...', Array)
#2 /var/www/MISP/app/Lib/cakephp/lib/Cake/Model/Datasource/DboSource.php(1026): DboSource->execute('INSERT INTO `mi...')
```

Seems to work though. Just an initial 500.

**Comment by iglocska** [comment]
Let's stick to the mode staying enabled for now, will test it once I am
back home and change galaxyclusters to comply with it. Guess my laptop only
has MySQL 5.6 running so I didn't catch it. Thanks for the heads-up!

**Comment by xme** [comment]
Facing the same issue :(

**Comment by FloatingGhost** [comment]
@xme there's a fix at the top of the issue - use that in the meantime

**Comment by iglocska** [comment]
Sorry for the delays, I thought I would be able to do it last night after flying back home, but I just keeled over and slept :) Just git pull and you'll have the fixes to both the MySQL 5.7 issue and the missing default for description that @xme described.
